core/server/objects.go

```diff
@@ -0,0 +1,44 @@
+package server
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/weaveworks/weave-gitops/core/server/types"
+	pb "github.com/weaveworks/weave-gitops/pkg/api/core"
+	"github.com/weaveworks/weave-gitops/pkg/server/auth"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (cs *coreServer) GetObject(ctx context.Context, msg *pb.GetObjectRequest) (*pb.GetObjectResponse, error) {
+	clustersClient, err := cs.clientsFactory.GetImpersonatedClient(ctx, auth.Principal(ctx))
+	if err != nil {
+		return nil, fmt.Errorf("error getting impersonating client: %w", err)
+	}
+
+	gvk := types.GetGVK(msg.Kind)
+	if gvk == nil {
+		return nil, fmt.Errorf("Looking up objects of type %v not supported", msg.Kind)
```

Comment:
To reduce the future chances for us to get `nil reference`, we should make GetGVK return an error too.  

This error should belong to the GetGVK function and we could just propagate the error here or wrap it like above.